### PR TITLE
DBCP-573 DataSource implementations do not implement Wrapper interface correctly

### PR DESCRIPTION
**Pull request by reda-alaoui**

### CODE REVIEW

**Comment by garydgregory** [comment]
> Hello?

Volunteer, busy.

File: src/main/java/org/apache/commons/dbcp2/BasicDataSource.java [hunk 1]
```diff
@@ -2445,8 +2445,12 @@ protected void startPoolMaintenance() {
     }
 
     @Override
+    @SuppressWarnings("unchecked")
     public <T> T unwrap(final Class<T> iface) throws SQLException {
-        throw new SQLException("BasicDataSource is not a wrapper.");
+        if (iface.isInstance(this)) {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/dbcp2/BasicDataSource.java:2450]: Why not reuse `isWrapperFor()`?

Comment (reda-alaoui) [reply src/main/java/org/apache/commons/dbcp2/BasicDataSource.java]: Yes that is better.

**Comment by garydgregory** [comment]
@reda-alaoui 
May you please resolve conflicts with the master branch?
TY.

**Comment by reda-alaoui** [comment]
@garydgregory conflicts resolved :)

**Pull request merged**